pci/src/vfio.rs

```diff
@@ -433,7 +446,7 @@ impl VfioCommon {
                     bar_addr = allocator
                         .lock()
                         .unwrap()
-                        .allocate_io_addresses(None, region_size, Some(0x4))
+                        .allocate_io_addresses(base_addr, region_size, Some(0x4))
```

Comment:
> Do we care about the case that the actual returned `bar_addr` is different from the requested `base_addr`?

The new `bar_addr` is used.

Reply:
> Do we care about the case that the actual returned `bar_addr` is different from the requested `base_addr`?

Of course we do, it shouldn't happen, and that's ensured directly by `allocate_io_address()` which relies on `allocate()`. If `base_addr` is `Some`, and if the available range can't be found, it will return `None` since it couldn't honor the request, meaning Cloud Hypervisor will throw the error `PciDeviceError::IoAllocationFailed`.

Reply:
Now I see it. Thanks for the detailed explanation.